tests/test_restarter.py

```diff
@@ -191,7 +192,7 @@ async def test_async_restart_check(config, install_kernel, debug_logging):
     km = AsyncIOLoopKernelManager(kernel_name=install_kernel, config=config)
 
     cbs = 0
-    restarts = [asyncio.Future() for i in range(N_restarts)]
+    restarts = [asyncio.futures.Future() for i in range(N_restarts)]
```

Comment:
This should still be `asyncio.Future`. That's the correct, public name for this. `asyncio.futures` is an implementation detail, which can change.

Reply:
We needed to use concurrent futures in the non-async tests, and asyncio futures in the async ones, I'll update to use `asyncio.Future`